x-pack/plugins/infra/public/common/alerts/create_alerts_es_query.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { getTime } from '@kbn/data-plugin/common';
+import { ALERT_TIME_RANGE } from '@kbn/rule-data-utils';
+import { buildEsQuery, Filter } from '@kbn/es-query';
+import type { HostsState } from '../../pages/metrics/hosts/hooks/use_unified_search_url_state';
+import type { AlertStatus } from '../../pages/metrics/hosts/types';
+import { ALERT_STATUS_QUERY } from './constants';
+import { buildCombinedHostsFilter } from '../../utils/filters/build';
+import type { AlertsEsQuery } from '../../pages/metrics/hosts/hooks/use_alerts_query';
```

Comment:
Could this one be moved to `common/alerts` too?

Reply:
Moved, thanks for spotting that!